src/api/app/controllers/webui/packages/build_log_controller.rb

```diff
@@ -5,53 +5,29 @@ class BuildLogController < Webui::WebuiController
       include Webui::NotificationsHandler
 
       before_action :check_ajax, only: :update_build_log
+      before_action :set_project
+      before_action :set_repository
+      before_action :set_architecture
       before_action :check_build_log_access
```

Comment:
```suggestion
      before_action :set_project
      before_action :check_build_log_access
      before_action :set_repository
      before_action :set_architecture
```
We could check for permissions (access) just after setting the project, before setting the repository and the architecture.

Reply:
We could but `check_build_log_access` is dropped in the last commit

Reply:
I was aware.

We could move the `before_action :set_object_to_authorize` before setting the repository and the architecture.

Reply:
To prevent for setting unnecessary data in the case that permissions are not granted. 

First we check for permissions with the minimal amount of queries needed. If permission is granted, we query the rest of the data (repositories and architectures).

Reply:
Not that repository or architecture are not required too to do anything but okay :)